src/graph/serializers/serialize_graph.c

```diff
@@ -293,22 +319,25 @@ void _RdbSaveEdges(RedisModuleIO *rdb, const Graph *g, Schema *s) {
         while(true) {
             bool depleted = false;
             GxB_MatrixTupleIter_next(it, &dest, &src, &depleted);
+            e.srcNodeID = src;
+            e.destNodeID = dest;
+
             if(depleted) break;
 
             GrB_Matrix_extractElement_UINT64(&edgeID, M, dest, src);
-            // Edge ID.
-            RedisModule_SaveUnsigned(rdb, edgeID);
-            // Source node ID.
-            src = _updatedID(g->nodes->deletedIdx, src);
-            RedisModule_SaveUnsigned(rdb, src);
-            // Destination node ID.
-            dest = _updatedID(g->nodes->deletedIdx, dest);
-            RedisModule_SaveUnsigned(rdb, dest);
-            // Relation type.
-            RedisModule_SaveUnsigned(rdb, r);
-            // Edge properties.
-            Graph_GetEdge(g, edgeID, &e);
-            _RdbSaveEntity(rdb, e.entity, edge_attribute_mapping);
+            if(SINGLE_EDGE(edgeID)) {
+                edgeID = SINGLE_EDGE_ID(edgeID);
+                Graph_GetEdge(g, edgeID, &e);
+                _RdbSaveEdge(rdb, g, &e, r, edge_attribute_mapping);
+            } else {
+                EdgeID *edgeIDs = (EdgeID*)edgeID;
+                int edgeCount = array_len(edgeIDs);
+                for(int i = 0; i < edgeCount; i++) {
+                    edgeID = edgeIDs[i];
+                    Graph_GetEdge(g, edgeID, &e);
+                    _RdbSaveEdge(rdb, g, &e, r, edge_attribute_mapping);
+                }
+            }
```

Comment:
While this doesn't break compatibility at all, I think we should still increment `GRAPHCONTEXT_TYPE_ENCODING_VERSION`, since loading an RDB into an older version will delete all but one of the edges that share a src-dest-reltype.

Reply:
👍